tests/experimental/pydantic/test_basic.py

```diff
@@ -187,6 +189,94 @@ class UserType:
     assert field2.type is GroupType
 
 
+def test_referencing_other_registered_models_register_nested():
+    # Automatically register with register_nested
+    class Name(pydantic.BaseModel):
+        value: str
+
+    class Group(pydantic.BaseModel):
+        name: Name
+
+    class User(pydantic.BaseModel):
+        age: int
+        group: Group
+
+    @strawberry.experimental.pydantic.type(
+        User, register_nested=RegisterNestedOptions()
+    )
+    class UserType:
+        age: strawberry.auto
+        group: strawberry.auto
+
+    definition: TypeDefinition = UserType._type_definition
+    assert definition.name == "UserType"
+
+    [field1, field2] = definition.fields
+
+    assert field1.python_name == "age"
+    assert field1.type is int
+
+    assert field2.python_name == "group"
+    assert field2.type.__name__ == "Group", "should not rename type by default"
```

Comment:
For consistency with the rest of the tests in the repo, can we remove the custom assert messages? If you thin the message is important, let's just make it a comment/docstring on the test.

Reply:
sure